game-core/src/main/java/org/triplea/live/servers/ServerYamlParser.java

```diff
@@ -0,0 +1,58 @@
+package org.triplea.live.servers;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.snakeyaml.engine.v1.api.Load;
+import org.snakeyaml.engine.v1.api.LoadSettingsBuilder;
+import org.triplea.util.Version;
+
+public class ServerYamlParser implements Function<InputStream, LiveServers> {
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public LiveServers apply(final InputStream inputStream) {
+    final Load load = new Load(new LoadSettingsBuilder().build());
+    final Map<?, ?> yamlProps;
+    try {
+      yamlProps = (Map<?, ?>) load.loadFromInputStream(inputStream);
+    } catch (final ClassCastException e) {
+      throw new IllegalArgumentException("Invalid yaml format");
+    }
+
+    if (yamlProps == null) {
+      throw new IllegalArgumentException("Invalid yaml file");
+    }
+
+    final String latest =
+        (String)
+            Optional.ofNullable(yamlProps.get("latest"))
```

Comment:
I don't want to get into the formatting argument here, but I think it's cleaner to do the string casting inside `Optional#ofNullable`

Reply:
As a general comment, whenever I find myself prefacing a review statement with something like:
> I don't want to get into the formatting argument here,

I would wonder to myself if the issue is important enough to be worth commenting on. The goal of CR is not to find and perfect every line detail, but to be sure there is reasonable clarity, no errors, design and technical debt are reasonable. One thing I've personally struggled with in reviews is to start at a high level first and then go low-level last.

With that said, I've no real preference one way or another, inside does have some benefits, not a bad clean-up; updated: c03e66b01

Reply:
To continue debating with myself, and unsolicited advice, @RoiEXLab happy to get the code cleaner, let's be sure to not lose sight of the forest and be sure the high level is right.